### PR TITLE
Feature: Update Component Props & Type Dotting Component Props

### PR DESCRIPTION
**Pull request by hunkim98**
🚀 [Related Issue: #18 ]

### Preview

Screen Shot 2023-05-14 at 12 07 13 PM

<br/>

## Remove `initBrushColor` & `initBrushTool` from Dotting props. Instead use `brushColor` & `brushTool`

- **[🎨Component] Remove `initBrushColor` & `initBrushTool`. Instead use `brushColor` & `brushTool`**

  - `initBrushColor` & `initBrushTool` seemed to complicate Dotting props. I instead changed the props to `brushColor` & `brushTool`. Now the user can use `useState` to control `brushColor` & `brushTool`.
  - The below code is an example on how you could use useState for controling brush tool and brush color
```ts
import { Dotting, BrushTool } from "dotting";
import { useState } from "react"

const DottingExample = () => {
  const [brushTool, setBrushTool] = useState<BrushTool>(BrushTool.DOT);
  const [brushColor, setBrushColor] = useState<string>(brushColor);
  return (
    <div>
      <Dotting width={300} height={300} brushColor={brushColor} brushTool={brushTool} />
    </div>
  );
};
```
  
- **[📒Docs] Type `Dotting` component props**

  - Previously the `Dotting` component argTypes did not sync with recent updates on `Dotting` props.
  - I created several functions to manage argTypes easily. I have created helper functions such as `generateComponentControlForEnum` & `generateComponentControl`

- **[🔗Other] Fix `backgroundMode` change not applying**

  - While working on the props, I figured that `backgroundMode` change did not apply. 
  - I fixed the issue by fixing the error.

<br/>

## Notes

- Since the props of `Dotting` has changed and major changes have been applied to Dotting, a new version `1.1.0` will be released if this PR is merged. @lerrybe @Lee-Si-Yoon 

<br/>

## Next Up?

- For the ease of comparing what I have done, I have set the branch to merge as `feature/selectTool2`. After people have confirmed the changes, I will change the base branch to merge to the `main` branch

### CODE REVIEW

**Comment by hunkim98** [comment]
I will merge this request tonight or tomorrow morning to update our `dotting` version to `1.1.0`. @lerrybe @Lee-Si-Yoon

File: src/components/Dotting.tsx [hunk 1]
```diff
@@ -30,14 +30,17 @@ export interface DottingProps {
   gridStrokeWidth?: number;
   isGridVisible?: boolean;
   backgroundMode?: "checkerboard" | "color";
-  backgroundColor?: React.CSSProperties["color"];
+  backgroundColor?: string;
```
Comment (Lee-Si-Yoon) [inline src/components/Dotting.tsx:33]: is regular string better than `React.CSSProperties["color"]`?

File: stories/Dotting.stories.tsx [hunk 2]
```diff
@@ -1,13 +1,104 @@
...
+  generateComponentControl,
+  generateComponentControlForEnum,
+} from "./utils/componentControl";
+import { KeysEnum, StoriesComponentItem } from "./utils/types";
+import { BrushTool } from "../src/components/Canvas/types";
+import DottingComponent, { DottingProps } from "../src/components/Dotting";
 
 //blog.harveydelaney.com/creating-your-own-react-component-library/
 
+const DottingComponentArgTypes: KeysEnum<
```
Comment (Lee-Si-Yoon) [inline stories/Dotting.stories.tsx:13]: this is great!

**Review by Lee-Si-Yoon — APPROVED** [review]

File: src/components/Dotting.tsx (hunk 1 above)
Comment (hunkim98) [reply src/components/Dotting.tsx:33]: I tried using react.cssproperties but it seems that stories.tsx does not interpret it well (when cssproperties is used, there is no color select option for the dotting component stories page) If string is used, the docs allows you to change the color of the property.

**Review by lerrybe — APPROVED** [review]

**Pull request merged**